pilopt/src/inliner.rs

```diff
@@ -50,26 +37,12 @@ fn try_apply_substitution<T: FieldElement, V: Ord + Clone + Hash + Eq + Display>
         let constraint = &constraint_system.algebraic_constraints[idx];
 
         for (var, expr) in find_inlinable_variables(constraint) {
-            if keep.contains(&var) {
-                continue;
-            }
-
-            if is_valid_substitution(
-                &var,
-                &expr,
-                &constraint_system.algebraic_constraints,
-                max_degree,
-            ) {
+            if is_valid_substitution(&var, &expr, constraint_system, max_degree) {
                 log::debug!("Substituting {var} = {expr}");
                 log::debug!("  (from identity {constraint})");
-                constraint_system
-                    .algebraic_constraints
-                    .iter_mut()
-                    .enumerate()
-                    .filter(|(i, _)| *i != idx)
```

Comment:
Note that now we also replace in the original constraint. I think this would typically lead to it being `0 = 0` afterwards. But in any case, it ends up being removed.